app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java

```diff
@@ -242,7 +244,25 @@ private void saveSharedPreferencesToFile(final File dst) {
     private void importDatabase(final String filePath) {
         // check if file is supported
         try (ZipFile zipFile = new ZipFile(filePath)) {
-        } catch (final IOException ioe) {
+            final Enumeration<? extends ZipEntry> entries = zipFile.entries();
+            boolean containsDbFile = false;
+            boolean containsSettingsFile = false;
+
+            while (entries.hasMoreElements()) {
+                final ZipEntry entry = entries.nextElement();
+                if (entry.getName().equals("newpipe.db")) {
```

Comment:
Maybe you could also check if `entry.getSize() > 0` here, but I am not sure more checks are needed: if the user manually provides an invalid zip which seems legal it is expected for the app to show an error report. Instead, in case something unexpected happened during exporting, resulting in corrupted files, I also think it would be a good idea to show an error report with information, instead of just showing "invalid zip"

Reply:
Checking the export doesn't ensure that the user picks these "previously proved" files, does it? How about a `Snackbar` with an action to show the error? (Similar to the case when comments couldn't be loaded)

Reply:
Yeah, I'd prefer a `Snackbar` with a custom message based on what caused the error

Reply:
Okay, I'll update it then soon.